pkg/catalog/catalog.go

```diff
@@ -66,9 +65,94 @@ func Exists(client *occlient.Client, componentType string, componentVersion stri
 	return true, nil
 }
 
+// ListServices lists all the available service types
+func ListServices(client *occlient.Client) (ServiceTypeList, error) {
+
+	clusterServiceClasses, err := getClusterCatalogServices(client)
+	if err != nil {
+		return ServiceTypeList{}, errors.Wrapf(err, "unable to get cluster serviceClassExternalName")
+	}
+
+	// Sorting service classes alphabetically
+	// Reference: https://golang.org/pkg/sort/#example_Slice
+	sort.Slice(clusterServiceClasses, func(i, j int) bool {
+		return clusterServiceClasses[i].Name < clusterServiceClasses[j].Name
+	})
+
+	return ServiceTypeList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceTypeList",
+			APIVersion: "odo.openshift.io/v1alpha1",
+		},
+		Items: clusterServiceClasses,
+	}, nil
+}
+
+// SearchService searches for the services
+func SearchService(client *occlient.Client, name string) (ServiceTypeList, error) {
+	var result []ServiceType
+	serviceList, err := ListServices(client)
+	if err != nil {
+		return ServiceTypeList{}, errors.Wrap(err, "unable to list services")
+	}
+
+	// do a partial search in all the services
+	for _, service := range serviceList.Items {
+		if strings.Contains(service.ObjectMeta.Name, name) {
+			result = append(result, service)
+		}
+	}
+
+	return ServiceTypeList{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "ServiceTypeList",
+			APIVersion: "odo.openshift.io/v1alpha1",
+		},
+		Items: result,
+	}, nil
+}
+
+// getClusterCatalogServices returns the names of all the cluster service
+// classes in the cluster
+func getClusterCatalogServices(client *occlient.Client) ([]ServiceType, error) {
+	var classNames []ServiceType
+
+	classes, err := client.GetClusterServiceClasses()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to get cluster service classes")
```

Comment:
Is this error message going to show up when `odo catalog list services` fails? I'm kind of confused as to what a `cluster service class` means and am wondering if the users might end up in similar situation as well.

Reply:
Nope this is normal. This was from previous code.